server/channels/app/post.go

```diff
@@ -700,7 +700,7 @@ func (a *App) UpdatePost(c *request.Context, post *model.Post, safeUpdate bool)
 	pluginContext := pluginContext(c)
 	a.ch.RunMultiHook(func(hooks plugin.Hooks) bool {
 		newPost, rejectionReason = hooks.MessageWillBeUpdated(pluginContext, newPost.ForPlugin(), oldPost.ForPlugin())
-		return post != nil
```

Comment:
So I guess this never worked?  https://github.com/mattermost/mattermost-plugin-profanity-filter/blob/a4bfa2172aadd6877880a61cb566ac26582bc8d8/server/plugin.go#L54C10-L54C13

Reply:
But somehow in the demo in the Readme, it does work https://github.com/mattermost/mattermost-plugin-profanity-filter so I'm confused 🤔

Reply:
In those, they are creating a new message, not updating an existing message.

Reply:
Ah ok, so the one that doesn't work is the Edit because that is also called here: https://github.com/mattermost/mattermost-plugin-profanity-filter/blob/a4bfa2172aadd6877880a61cb566ac26582bc8d8/server/plugin.go#L72 

With you change then I guess it will start working again

Reply:
Yes. I have double checked with the plugin on master and the plugin does it well when we don't ask to reject the messages (it just censor the word). But if we tell the plugin to reject the messages, the update won't be rejected and will leave the swear word uncensored.